### PR TITLE
Include ELA example

### PR DESCRIPTION
**Pull request by cabelo**
### This pullrequest changes

This example performs an Error Level Analysis in the image provided using the same technique as the FotoForensics website. ELA allows to see visually the changes made in a JPG image based in it's compression error analysis.

See my post in my blog: https://assuntonerd.com.br/2018/06/06/analise-forense-de-imagem-com-visao-computacional/

https://cabelovivaolinux.files.wordpress.com/2018/06/screenshot_20180625_013246.png

https://cabelovivaolinux.files.wordpress.com/2018/06/screenshot_20180625_013227.png

### CODE REVIEW

File: samples/cpp/ela.cpp [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+            ptr_out[2] = abs(ptr_input[2] - ptr_compress[2]) * scale_value;
+
+            ptr_input += 3;
+            ptr_compress += 3;
+            ptr_out += 3;
+        }
+    }
+
+    // Shows processed image
+    cv::imshow("Error Level Analysis", output);
```
Comment (vpisarev) [inline samples/cpp/ela.cpp]: since you already put `using namespace cv`, please, remove all `cv::`

File: samples/cpp/ela.cpp [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    if (compressed_img.empty())
+    {
+        std::cout << "> Error in load file" << std::endl;
+        exit(EXIT_FAILURE);
+    }
+
+    Mat output = Mat::zeros(image.size(), CV_8UC3);
+
+    // Compare values through matrices
+    for (int row = 0; row < image.rows; ++row)
```
Comment (vpisarev) [inline samples/cpp/ela.cpp]: replace this loop with `absdiff` and `convertTo`

File: samples/cpp/ela.cpp [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+Mat compressed_img;
+
+static void processImage(int , void* )
+{
+    // Compression jpeg
+    std::vector<int> compressing_factor;
+    compressing_factor.push_back(CV_IMWRITE_JPEG_QUALITY);
+    compressing_factor.push_back(quality);
+
+    imwrite("temp.jpg", image, compressing_factor);
```
Comment (vpisarev) [inline samples/cpp/ela.cpp]: do not write compressed image to a file, use `encode` and `decode` functions from imgcodecs.

File: samples/cpp/ela.cpp [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+} 
+
+int main (int argc, char* argv[])
+{
+
+    CommandLineParser parser(argc, argv, keys);
+    if(argc == 1 || parser.has("help"))
+    {
+        parser.printMessage();
+	std::cout<<std::endl<<"Example: "<<std::endl<<argv[0]<< " -input=../../data/ela_modified.jpg"<<std::endl;
```
Comment (vpisarev) [inline samples/cpp/ela.cpp]: can we use some other image, like lena or fruits or baboon? the provided image is too dark and the interesting texture is only in a small part of it

File: samples/cpp/ela.cpp [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    // Check image
+    if (!image.empty())
+    {
+        cv::namedWindow("Error Level Analysis");
+        cv::imshow("Error Level Analysis", image);
+        cv::createTrackbar("Scale", "Error Level Analysis", &scale_value, 100, processImage);
+        cv::createTrackbar("Quality", "Error Level Analysis", &quality, 100, processImage);
+    }
+
+    cv::waitKey(0);
```
Comment (vpisarev) [inline samples/cpp/ela.cpp]: if there is no image, the app should quit immediately, not wait for a key pressed

File: samples/cpp/ela.cpp (hunk 4 above)
Comment (cabelo) [reply samples/cpp/ela.cpp]: This example show the region modified in photoshop (linux logo). so I added these images.

**Comment by vpisarev** [comment]
:+1:

**Pull request merged**